eng/tox/tox.ini

```
@@ -91,8 +113,8 @@ setenv =
 deps =
   -rdev_requirements.txt
 commands =
-    python -m pip install pylint=={[testenv:pylint]pylint_version}
-    python -m pip install azure-pylint-guidelines-checker==0.5.6 --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+    {[tox]pip_command} install pylint=={[testenv:pylint]pylint_version}
+    {[tox]pip_command} install azure-pylint-guidelines-checker==0.5.6 --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/" --index-strategy=unsafe-best-match
```

```suggestion
    {[tox]pip_command} install azure-pylint-guidelines-checker==0.5.6 --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/" --index-strategy=unsafe-best-match
```
```suggestion
    {[tox]pip_command} install azure-pylint-guidelines-checker==0.5.6 --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/" **--index-strategy=unsafe-best-match**
```

this seems like it could end up being --extra-index-url but uv doesn't support index-url the same way as pip so this is required if we wnat to grab it not off of pypi.